Create 'Provide a good engineering experience' principle (#186)
HO-CTO/engineering-guidance-and-standards: docs/principles/provide-good-engineering-experience.md

---
layout: sub-navigation
order: 1
title: Provide a good engineering experience
date: git Last Modified
tags:
- Ways of working
---

{% import "macros/tags.njk" as pageTag %}

Last updated: {{ page.date | postDate }}
{{ pageTag.tags(tags)  }}

At the Home Office, we recognise that the [engineer experience](https://github.blog/2023-06-08-developer-experience-what-is-it-and-why-should-you-care/) is comprised of many different aspects, and that getting those things right allows engineers to produce their best work.  We will invest in the continuous improvement of our engineering experience.

---

## Rationale

> The Home Office engineering experience is more than tooling and platforms and technology meeting engineers' needs. It is affected by our working relationships, our community, our structures and our culture. Engineers complete work more quickly, and at higher quality, when the tools, environments and teams they work with provide the right capabilities and support. When they don't have those things, work moves more slowly, technical debt is accrued and security can be compromised. We want to reduce day to day friction and duplication, to improve productivity.

Reviewer: ```suggestion
The Home Office engineering experience is more than tooling and platforms and technology meeting engineers' needs. It is affected by our working relationships, our community, our structures and our culture. Engineers complete work quicker and at higher quality when the tools, environments and teams they work with provide the right capabilities and support. When they don't have those things, work moves more slowly, technical debt is accrued and security can be compromised. We want to reduce day-to-day friction and duplication, to improve productivity.
```

It is important to remember that engineers are users too, we already have proven patterns for the best ways to meet user needs and they should be followed. To improve engineer experience we will meet engineering needs in the same way as we would with any other user need, centering engineers' feedback and experience when testing solutions that have been designed for them. This will provide the Home Office with a measurable and evidenced approach to factors that contribute to productivity and efficiency.

---

## Applications and Implications

> - Approach cloud platform engineering in a way that delivers platforms as products that are developed according to tenants needs

Reviewer 2: not sure about the word Tenant here.. Either this is about engineers, or is about tenants, so maybe this point is in platform principles not here?

Reviewer: Agree this should be written from an engineer's perspective, maybe:
```suggestion
- The Home Office's cloud platform engineering should deliver platforms as products that are developed according to tenant projects' needs. Engineers should engage with platforms to provide the feedback needed to continuously improve the developer experience.
```

> - Do user research with engineers to understand how they use tools and services, and where we can improve

Reviewer 2: remove 'user' here. avoid conflation/confusion with big U User Research which I think is different

Reviewer: I would argue that the engineers are the users of the tools and services and using user research methodology is appropriate.

Author: Accept that the sentence needs to be clarified, but agree with Jeff on the intent. Will reword

> - The Home Office's cloud platform engineering should deliver platforms as products that are developed according to tenant projects' needs. Engineers should engage with platforms to provide the feedback needed to continuously improve the developer experience.

Reviewer 3: ```suggestion
- The Home Office's cloud platform engineering should deliver platforms as products that are developed according to tenant projects' needs. Engineers should engage with platforms to provide the feedback needed to continuously improve the developer experience
```

So it is consistent with the other bullet points, remove the full stop
